Automattic/wp-calypso
# Uses a custom script to detect npm vs yarn

#### Changes proposed in this Pull Request

* Replaces 'check-npm-client' with a custom tool that does something similar, but doesn't depend on any external package.

#### Testing instructions

* Checkout the branch and delete `node_modules`.
* Try `npm run start`, it should throw an error asking you to use `yarn` instead. Verify that `yarn run start` does not throw the error.
* Create an empty file `package-lock.json`and delete `yarn.lock` and try `yarn run start`.  It should throw an error asking you to use `npm` instead. Verify that `npm run start` does not throw the error.
* Repeat the steps above but using `(npm|yarn) install` instead of `(npm|yarn) run start`
* Repeat the steps above, but using `(npm|yarn) dev` inside `apps/full-site-editing` (empty `package-lock.json` and delete existing `yarn.lock` in the project's root)
